### PR TITLE
WIP: CI with kind

### PR DESCRIPTION
**Pull request by clkao**
Related to #1206

### CODE REVIEW

File: ci/test.sh [hunk 1]
```diff
@@ -4,22 +4,33 @@ set -eux
...
+    IP=$(/sbin/ifconfig $iface | grep 'inet addr' | cut -d: -f2 | awk '{print $1}');
     if [ -n "$IP" ]; then
         echo "IP: $IP"
         break
     fi
 done
 if [ -z "$IP" ]; then
     echo "Failed to get IP, current interfaces:"
-    ifconfig -a
+    /sbin/ifconfig -a
```
Comment (minrk) [inline ci/test.sh]: was `/sbin/` not on PATH already? I think it had to be in order to be working.

File: ci/install-kind.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+set -ex
+
+mkdir -p bin
```
Comment (minrk) [inline ci/install-kind.sh]: There's lots of duplication now between install-kind and install-minikube. 

I don't think we need to run tests in both minikube and kind. I think we should either switch to kind or not at all. Does that make sense? What is the value of maintaining and testing multiple bootstrap environments?

Comment (clkao) [reply ci/install-kind.sh]: Right, that was also my idea to incrementally remove minikube once we are happy with kind.

However the unreleased kind 0.3.0 switched to the containerd CRI runtime, as suppose to using docker runtime.  I want to bring this up to see if people are comfortable when the CI runs solely on a non-typical kubernetes environment of underlying runtime.

File: ci/test.sh (hunk 1 above)
Comment (clkao) [reply ci/test.sh]: it didn't seem to work in the local environment created by Vagrantfile

**Comment by consideRatio** [comment]
@clkao :heart: thank you for working on this! I'm catching up with kind now!

I'm trying out to use kind as a local dev environment while learning to use it.

I don't see any mention of using `kind load docker-image` in this, yet the test succeed... Hmmmm.... I was forced to do `kind load docker-image jupyterhub/k8s-hub` in order to make the docker image built by chartpress to be available locally.

Is this because this PR didn't trigger chartpress to try rebuild the hub image for example or similar? Hmm... I'm not at all confident about this...

**Comment by consideRatio** [comment]
Relevant: https://github.com/kind-ci/examples/blob/master/.travis.yml

File: ci/install-kind.sh (hunk 2 above)
Comment (consideRatio) [reply ci/install-kind.sh]: I'm 100% comfortable for z2jh CI to run with kind!

Comment (manics) [reply ci/install-kind.sh]: I've just tried Kind locally (not using these scripts), it's really easy to get running on a Linux system. Main downside so far is I can't get NetworkPolicies to work.

**Comment by manics** [comment]
In case anyone's interested I tried and failed to get NetworkPolicies working on Kind:
- https://github.com/kubernetes-sigs/kind/issues/842
- https://docs.projectcalico.org/v3.9/getting-started/kubernetes/installation/calico
```yaml
# https://github.com/kubernetes-sigs/kind/blob/v0.5.1/site/content/docs/user/kind-example-config.yaml
# kind create cluster --config kind-config.yaml
kind: Cluster
apiVersion: kind.sigs.k8s.io/v1alpha3
networking:
  disableDefaultCNI: true
```
```sh
kind create cluster --config kind-config.yaml

curl https://docs.projectcalico.org/v3.9/manifests/calico.yaml | sed -e "s?192.168.0.0/16?10.244.0.0/24?g" | kubectl apply -f -

kubectl wait --for=condition=Ready node/kind-control-plane --timeout=5m
```

File: ci/install-kind.sh [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+# nsenter is included on xenial
+
+# install socat (required by helm)
+sudo apt-get update && sudo apt-get install -y socat
```
Comment (consideRatio) [inline ci/install-kind.sh]: It is? But this was not added to the minikube script? I tracked down this to debug why it took so much longer to run the jobs with kind compared to using minikube.

**Comment by consideRatio** [comment]
@manics @clkao I got the network policies running functionally in #1422!

**Comment by manics** [comment]
@consideRatio That's awesome! I'll check it out as soon as I get time.

**Pull request closed without merging**